lnwallet/rpcwallet/sign_coordinator.go

```diff
@@ -0,0 +1,886 @@
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) *signrpc.MuSig2SignResponse {
+			return resp.GetMuSig2SignResponse()
+		},
+	)
+}
+
+// SignMessage sends a SignMessageReq to the remote signer and waits for the
+// corresponding response.
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) SignMessage(_ context.Context,
+	in *signrpc.SignMessageReq,
+	_ ...grpc.CallOption) (*signrpc.SignMessageResp, error) {
+
+	req := &walletrpc.SignCoordinatorRequest_SignMessageReq{
+		SignMessageReq: in,
+	}
+
+	return processRequest(
+		s, noTimeout,
+		func(reqId uint64) walletrpc.SignCoordinatorRequest {
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) *signrpc.SignMessageResp {
+			return resp.GetSignMessageResp()
+		},
+	)
+}
+
+// SignPsbt sends a SignPsbtRequest to the remote signer and waits for the
+// corresponding response.
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) SignPsbt(_ context.Context,
+	in *walletrpc.SignPsbtRequest,
+	_ ...grpc.CallOption) (*walletrpc.SignPsbtResponse, error) {
+
+	req := &walletrpc.SignCoordinatorRequest_SignPsbtRequest{
+		SignPsbtRequest: in,
+	}
+
+	return processRequest(
+		s, noTimeout,
+		func(reqId uint64) walletrpc.SignCoordinatorRequest {
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) *walletrpc.SignPsbtResponse {
+			return resp.GetSignPsbtResponse()
+		},
+	)
+}
+
+// processRequest is a generic function that sends a request to the remote
+// signer and waits for the corresponding response. If a timeout is set, the
+// function will limit the execution time of the entire function to the
+// specified timeout. If it is not set, configured timeouts will be used for
+// the individual operations within the function.
+func processRequest[R comparable](s *SignCoordinator, timeout time.Duration,
+	generateRequest func(uint64) walletrpc.SignCoordinatorRequest,
+	extractResponse func(*signerResponse) R) (R, error) {
+
+	var zero R
+
+	done, err := s.registerRequest()
+	if err != nil {
+		return zero, err
+	}
+	defer done()
+
+	startTime := time.Now()
+
+	// If a timeout is enforced, we will wait for the connection using the
+	// specified timeout. Otherwise, we will wait for the connection using
+	// the configured connection timeout.
+	if timeout != 0 {
+		err = s.waitUntilConnectedWithTimeout(timeout)
+	} else {
+		err = s.WaitUntilConnected()
+	}
+
+	if err != nil {
+		return zero, err
+	}
+
+	reqID := s.nextRequestID.Add(1)
+	req := generateRequest(reqID)
+
+	cleanUpChannel := s.createResponseChannel(reqID)
+	defer cleanUpChannel()
+
+	log.Debugf("Sending a %t to the remote signer with request ID %d",
```

Comment:
nit: I think the type here needs to be `%T`, `%t` is for boolean values.

Reply:
Otherwise this will log something like this:
```
Sending a &{%!t(*signrpc.SignMessageReq=&{{{} [] [] <nil>} 0 [] [0 253 128 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 128 32 0 138 130 81 161 103 52 180 109 2 150 148 2 81 101 58 129 136 164 101 49 196 14 59 225 210 104 230 179 94 254 152 244 99 1 126 137 191 99 79 183 27 51 153 255 121 97 114 97 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 0xc00387a810 true false false [] []})} to the remote signer with request ID 49
```

Reply:
Ah TIL, thanks! Addressed :)